vignettes/adtr.Rmd

```diff
@@ -0,0 +1,599 @@
+```
+
+At this step, it may be useful to join `ADSL` to your `TR` domain.  Only the
+`ADSL` variables used for derivations are selected at this step.  The rest of
+the relevant `ADSL` variables would be added later.
+
+```{r eval=TRUE}
+adsl_vars <- vars(RANDDT)
+tr <- derive_vars_merged(
+  tr,
+  dataset_add = adsl,
+  new_vars = adsl_vars,
+  by_vars = vars(STUDYID, USUBJID)
+)
+```
+
+## Merge TR with TU and derive new variables {#merge}
+
+Depending on your company specifications you may want to merge `TU` with `TR`.
+As an example, here we are keeping the tumor location, and deriving a new
+variable for the tumor group site. You may want to process other variables like
+`ADT`.
+
+```{r}
+tr <- derive_vars_merged(
+  tr,
+  dataset_add = rename(tu, TRLNKID = TULNKID),
+  new_vars = vars(TULOC),
+  by_vars = vars(STUDYID, USUBJID, TRLNKID)
+) %>% mutate(
+  TULOCGR1 = if_else(
+    TULOC == "LYMPH NODE",
+    "NODAL",
+    "NON-NODAL"
+  )
+)
+```
+
+```{r, echo=FALSE}
+dataset_vignette(tr, display_vars = vars(USUBJID, TRLNKID, TULOC, TULOCGR1))
+```  
+
+
+In addition, you could create any additional new variables that are required
+downstream when deriving the required parameters. For example, here we include
+lesion ID expected and lesion ID assessed. (any other examples?)
+
+```{r}
+tr <- mutate(
+  tr,
+  LSEXP = TRLNKID,
+  LSASS = if_else(!is.na(TRSTRESN), TRLNKID, NA_character_)
+)
+```
+
+## Pre-processing of input records {#input}
+
+The next step involves company-specific pre-processing of records for the
+required input to downstream parameter derivations.  See examples below.
+
+### Partial Date Imputation and Deriving `ADT`, `ADTF`, `AVISIT` etc
+
+If your data collection allows for partial dates, you could apply a
+company-specific imputation rule at this stage when deriving `ADT`. For this
+example, here we impute missing day to earliest possible date.
+
+```{r}
+tr <- derive_vars_dt(
+  tr,
+  dtc = TRDTC,
+  new_vars_prefix = "A",
+  highest_imputation = "D",
+  date_imputation = "first"
+) %>%
+  derive_vars_dy(
+    reference_date = RANDDT,
+    source_vars = vars(ADT)
+  )
+```
+
+
+```{r, echo=FALSE}
+dataset_vignette(tr, display_vars = vars(USUBJID, RANDDT, TRLNKID, TRDTC, ADT, ADTF, ADY))
+```  
+
+### Re-labelling unscheduled visits
+
+If you have data collected at unscheduled visits, which are not identified as
+unique in your SDTM or you have company specific rules.  Below is an example of
+how it could be re-labelled. <add more information once I see the template code>
+
+```{r}
+tr <- mutate(
+  tr,
+  AVISIT = VISIT,
+  AVISITN = VISITNUM
+)
+```
+
+## Derive Parameters TLDn and TLPn {#parameters}
+
+As mentioned earlier, depending on your company specifications you may either
+create parameters for TLDn and TLPn in ADTR or similar dataset.
+
+Set the `AVALx/ANL01FL` variables.
+```{r}
+adtr <- tr %>%
+  mutate(
+    tmp_lesion_nr = str_sub(TRLNKID, 3),
```

Comment:
`str_sub(TRLNKID, 3) `is only repeated twice then you have to remove with a `select(-)`, so I see no benefit in defining. I would just have `str_sub(TRLNKID, 3)` twice, but this is personal preference.

I am sure I read Hadley Wickham state if it repeats at least three times then abstract, two is ok (I may be making this up though).

Reply:
I leave for @bundfussr to comment

Reply:
I agree that there is no benefit with respect to the length of the code. However, I think for maintenance it is better not repeating definitions like the lesion number. Then it is easier to adjust the code.
Therefore I have updated the code such that `tmp_lesion_nr` is derived only once.